contrib/babelfishpg_common/src/varbinary.c

```diff
@@ -874,6 +1036,49 @@ varcharbinary(PG_FUNCTION_ARGS)
 	PG_RETURN_BYTEA_P(result);
 }
 
+Datum
+nvarcharbinary(PG_FUNCTION_ARGS)
+{
+	VarChar    *source = PG_GETARG_VARCHAR_PP(0);
+	char	   *data = VARDATA_ANY(source);
+	char	   *rp;
+	size_t		len = VARSIZE_ANY_EXHDR(source);
+	int32		typmod = PG_GETARG_INT32(1);
+	bool		isExplicit = PG_GETARG_BOOL(2);
+	int32		maxlen;
+	bytea	   *result;
+	StringInfoData 		buf;
+
+	if (!isExplicit)
+		ereport(ERROR,
+				(errcode(ERRCODE_DATATYPE_MISMATCH),
+				 errmsg("Implicit conversion from data type nvarchar to "
+						"binary is not allowed. Use the CONVERT function "
+						"to run this query.")));
+
+	initStringInfo(&buf);
+	TsqlUTF8toUTF16StringInfo(&buf, data, len);
+	data = buf.data;
+	len= buf.len;
+
+	/* If typmod is -1 (or invalid), use the actual length */
+	if (typmod < 0)
```

Comment:
should be < `VARHDRSZ`

Reply:
why? as i made comment previously, we simply need typmod == -1

Reply:
typmod < `VARHDRSZ` we cannot even accommodate the header so we have to treat it as -1. We follow this in all the existing cast functions.

Reply:
Even postgres cast functions check typmod against `VARHDRSZ`